### PR TITLE
feat: add Chocolatey to install GNU make for Windows

### PR DESCRIPTION
**Pull request by lemeurherve**
Having 'make' will allow us to use a common Makefile in buildDockerAndPublishImage wether on a Linux or on a Windows agent.
Installing 'make' with Chocolatey as it's where the last make for Windows version is available.

Ref: https://github.com/jenkins-infra/helpdesk/issues/2873

### CODE REVIEW

File: scripts/windows-2019-provision.ps1 [hunk 1]
```diff
@@ -172,7 +178,18 @@ $downloads = [ordered]@{
...
         'cleanuplocal' = 'true'
-    }
+    };
+    'chocolatey' = @{
+        'url' = 'https://github.com/chocolatey/choco/releases/download/{0}/chocolatey.{0}.nupkg' -f $env:CHOCOLATEY_VERSION;
+        'local' = "$baseDir\chocolatey.nupkg";
+        'expandTo' = "$baseDir\chocolatey.tmp";
+        'postexpand' = {
+            & "$baseDir\chocolatey.tmp\tools\chocolateyInstall.ps1";
+            & "choco install make"; # -- version $env:MAKE_VERSION";
```
Comment (lemeurherve) [inline scripts/windows-2019-provision.ps1:188]:
```suggestion
            & "choco install make";
```

**Comment by lemeurherve** [comment]
Finally 😛 🎉 

image

Failing updatecli for chocolatey as it's not yet in provision-env.yaml in the principal branch.

File: updatecli/updatecli.d/chocolatey.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+pullrequests:
+  default:
+    kind: github
+    scmid: default
+    targets:
+      - updateVersion
+    spec:
+      labels:
+        - dependencies
```
Comment (lemeurherve) [inline updatecli/updatecli.d/chocolatey.yml:44]: I've put `dependencies` instead of `enhancement` here, WDYT?

**Review by dduportal — APPROVED** [review]
\o/

**Pull request merged**